src/test/java/com/commercetools/sync/products/utils/ProductReferenceReplacementUtilsTest.java

```diff
@@ -343,10 +343,11 @@ public void buildProductQuery_Always_ShouldReturnQueryWithAllNeededReferencesExp
 
         assertThat(categoryReferencePair).isNotNull();
 
-        final List<Reference<Category>> categoryReferencesWithKeys = categoryReferencePair.getCategoryReferences();
+        final Set<ResourceIdentifier<Category>> categoryReferencesWithKeys =
+            categoryReferencePair.getCategoryResourceIdentifiers();
         final CategoryOrderHints categoryOrderHintsWithKeys = categoryReferencePair.getCategoryOrderHints();
         assertThat(categoryReferencesWithKeys).hasSize(1);
-        assertThat(categoryReferencesWithKeys.get(0).getId()).isEqualTo(categoryId);
+        assertThat(categoryReferencesWithKeys.iterator().next().getId()).isEqualTo(categoryId);
```

Comment:
instead of `hasSize()` + `iterator().next().getId()).isEqualTo()` use convenient assertJ's `contains`, `containsExactly`, `containsOnly` and so on.

Same for assertions below.

Reply:
61d173a

Reply:
better, but i expected smth like:
```java
assertThat(categoryReferencesWithKeys)
            .extracting(ResourceIdentifier::getId)
            .containsExactlyInAnyOrder(categoryId);
```

So, that you are verifying exactly set of IDs, not comparing sets

Reply:
@andrii-kovalenko-ct to be honest, I never used this `.extracting` from assertJ, but now I am a fan: e3945e7
Thanks!

Reply:
nice :)